### DISCUSSION TITLE
How dangerous is it to use Regolith with Debian Testing?

### DISCUSSION THREAD

**Discussion by Screwtapello** [regolith-linux/regolith-desktop#940] [General]
Thanks for making Regolith available for Debian - I've been running my own i3+GNOME monster for a few years and really like the combination, but I'm looking for a way to migrate to Wayland and haven't been able to figure out how to get Sway and my distro-provided SSH agents to cooperate.

However, I'm running Debian Testing, rather than Debian Stable, to be a bit more up-to-date. I understand that Regolith can't make any promises about supporting a distro that changes literally every day, but do you think it's likely that installing Regolith on Debian Testing would be *dangerous*? Like, data-loss dangerous? Or should I just expect that sometimes I'll have to switch to a different desktop session and report issues, or at worst uninstall Regolith packages to allow apt to install Debian-provided updates?

**Comment by kgilmer** [comment]
Hi @Screwtapello 

> I've been running my own i3+GNOME monster for a few years 

haha, nice!

> do you think it's likely that installing Regolith on Debian Testing would be dangerous?

I think worst case would be that you may be in some session and suddenly you cannot open new applications or use other keybindings if a background automated updated brings in some change into `testing` that has a regression.  There is nothing I am aware of in Regolith that would cause data loss.  Of course, there are no safety guarantees provided by the software, which is a combination of existing components, bash scripts,and rust daemons.

There may be regressions in functionality in cases where some upstream change that fails to build and that component has to be disabled.  This is currently [the case for Sway](https://github.com/regolith-linux/voulage/blob/main/stage/testing/debian/testing/package-model.json#L27) in fact.  For more control, you may wish to only update from Regolith deb repos manually.

Additionally, if you're interested in contributing tests to exercising Debian Testing builds, that may also be a way of having more assurance of quality:

Test script (currently verifies only the Debian package metadata correctly allows the desktop package to be installed, but any test expressible in a bash script might be added here): https://github.com/kgilmer/test-debian-testing-action/blob/main/entrypoint.sh#L20

Called from (currently disabled): https://github.com/regolith-linux/voulage/blob/main/.github/workflows/test-desktop-installable2.yml#L30
